### PR TITLE
do not block during size command

### PR DESCRIPTION
**Pull request by Nonius**

### CODE REVIEW

File: src/storage/cloud_storage.rs [hunk 1]
```diff
@@ -73,6 +73,23 @@ fn item_to_metadata(item: Item) -> ObjectMetadata {
...
 }
 
+fn item_to_file_info(item: Item) -> Fileinfo<PathBuf, ObjectMetadata> {
+    Fileinfo {
+        path: PathBuf::from(item.name),
+        metadata: ObjectMetadata {
+            last_updated: match u64::try_from(item.updated.timestamp_millis()) {
+                Ok(timestamp) => SystemTime::UNIX_EPOCH.checked_add(Duration::from_millis(timestamp)),
+                _ => None,
+            },
```
Comment (wgfm) [inline src/storage/cloud_storage.rs]: Didn't test this, but I think you can go from Result to Option with `ok()` which discards the error:

```rust
u64::try_from(item.updated.timestamp_millis())
  .map(|timestamp| SystemTime::UNIX_EPOCH.checked_add(Duration::from_millis(timestamp)))
  .ok()
```

https://doc.rust-lang.org/std/result/enum.Result.html#method.ok

File: src/storage/cloud_storage.rs [hunk 2]
```diff
@@ -73,6 +73,23 @@ fn item_to_metadata(item: Item) -> ObjectMetadata {
...
+            is_file: true,
+            size: match item.size.parse() {
+                Ok(size) => size,
+                //TODO: return 450
+                _ => 0,
+            },
```
Comment (wgfm) [inline src/storage/cloud_storage.rs]: In this case, I think you can do `item.size.parse().or_else(0)`, rather than `match`.

Comment (Nonius) [reply src/storage/cloud_storage.rs]: The TODO will solve this

File: src/storage/cloud_storage.rs (hunk 1 above)
Comment (Nonius) [reply src/storage/cloud_storage.rs]: with this we have an Option<Option<u64>> and flatten() is unstable.

**Pull request merged**